### PR TITLE
P2TR support

### PR DESCRIPTION
**Pull request by octobocto**
- proto: add taproot to script types
- src: maybe add taproot support

### CODE REVIEW

**Comment by torkelrogstad** [comment]
Would love some tests here

File: src/main/kotlin/no/bb/deskriptor/service/Service.kt [hunk 1]
```diff
@@ -95,6 +95,7 @@ data class Script(private val script: ScriptType) {
         // operators, and replace this `when` block with something that returns
         // a list of operators. For another day.
         ScriptType.SCRIPT_TYPE_WPKH_NESTED -> "sh(wpkh"
+        ScriptType.SCRIPT_TYPE_P2TR -> "p2tr"
```
Comment (octobocto) [inline src/main/kotlin/no/bb/deskriptor/service/Service.kt]: have no idea if this is correct (or how to test it)

**Comment by torkelrogstad** [comment]
Test vectors look good, but they're failing

**Comment by octobocto** [comment]
> Test vectors look good, but they're failing

I'm very sure they're correct, but I think the taproot method is wrong. How can I figure out what operator to call?

wrong:
```
ScriptType.SCRIPT_TYPE_P2TR -> "sh(tr"
```
also wrong:
```
ScriptType.SCRIPT_TYPE_P2TR -> "tr"
```

**Comment by octobocto** [comment]
BADABOM!

**Pull request merged**